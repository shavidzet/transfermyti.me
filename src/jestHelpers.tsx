/* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-explicit-any,no-console */
import React from "react";
import {
  RenderOptions,
  RenderResult,
  render as testRender,
} from "@testing-library/react";
import { ThemeProvider } from "emotion-theming";
import { theme } from "./components/emotion";

// These are exported for convenience (so we don't have to add an additional import
// to the test file).
export { createEvent, fireEvent, RenderResult } from "@testing-library/react";
export { theme } from "./components/emotion";

/**
 * Renders the specified component with react-testing-library wrapped in a theme
 * provider so we don't have to do it explicitly for each component.
 */
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult {
  return testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
}

type LogMethod = "log" | "warn" | "error";

export class ConsoleToggle {
  private consoleMethod: (...args: any[]) => void = console.warn;
  private readonly logMethod: LogMethod;

  public constructor(logMethod: LogMethod) {
    this.logMethod = logMethod;
  }

  public onBefore(): void {
    this.consoleMethod = console[this.logMethod];
    console[this.logMethod] = jest.fn();
  }

  public onAfter(): void {
    console[this.logMethod] = this.consoleMethod;
  }
}
