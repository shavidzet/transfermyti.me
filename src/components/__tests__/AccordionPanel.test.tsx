import React from "react";
import { render, RenderResult } from "~/jestHelpers";
import AccordionPanel from "../AccordionPanel";

const setup = (
  propOverrides: any = {},
): { props: any; wrapper: RenderResult } => {
  const props = {
    children: "Test Children",
    rowNumber: 1,
    title: "Test Title",
    ...propOverrides,
  };

  const wrapper = render(<AccordionPanel {...props} />);

  return { props, wrapper };
};

describe("the <AccordionPanel> component", () => {
  test("matches its snapshot with valid props", () => {
    const { wrapper } = setup();

    expect(wrapper.container.firstElementChild).toMatchSnapshot();
  });
});
