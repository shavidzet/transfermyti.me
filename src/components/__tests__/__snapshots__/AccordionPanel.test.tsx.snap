// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <AccordionPanel> component matches its snapshot with valid props 1`] = `
.emotion-2 {
  margin: 0;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.25rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 1rem;
  padding: 0.75rem 1rem;
  text-align: left;
  width: 100%;
  background-color: rgb(255,255,255);
  border-radius: 0.375rem;
  box-shadow: 0 1px 5px 0 rgba(0,0,0,0.2), 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12);
  color: rgb(30,120,161);
}

.emotion-1:hover {
  background-color: rgb(30,120,161);
  color: rgb(255,255,255);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1:hover path {
  fill: rgb(255,255,255);
}

.emotion-0 {
  fill: rgb(30,120,161);
}

.emotion-3 {
  margin-bottom: 2rem;
  padding: 0 1rem;
  display: none;
}

<div>
  <h3
    class="emotion-2"
  >
    <button
      aria-controls="accordion-content-1"
      aria-expanded="false"
      class="emotion-1"
      id="accordion-title-1"
    >
      Test Title
      <svg
        aria-describedby="icon-circleAdd-desc"
        aria-labelledby="icon-circleAdd-title"
        height="32"
        role="presentation"
        version="1.1"
        viewBox="0 0 48 48"
        width="32"
      >
        <title
          id="icon-circleAdd-title"
        >
          Circle with Plus Sign Icon
        </title>
        <desc
          id="icon-circleAdd-desc"
        >
          A circle with a plus sign in the middle.
        </desc>
        <path
          class="emotion-0"
          d="M26 14h-4v8h-8v4h8v8h4v-8h8v-4h-8v-8zM24 4C12.95 4 4 12.95 4 24s8.95 20 20 20 20-8.95 20-20S35.05 4 24 4zm0 36c-8.82 0-16-7.18-16-16S15.188 24 8s16 7.18 16 16-7.18 16-16 16z"
          data-testid="svg-icon-path"
        />
      </svg>
    </button>
  </h3>
  <div
    aria-hidden="true"
    aria-labelledby="accordion-title-1"
    class="emotion-3"
    id="accordion-content-1"
    role="region"
  >
    Test Children
  </div>
</div>
`;
